src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -238,11 +238,13 @@ public static void writeAllSiteTypesToBuilder(Device device, DeviceResources.Dev
             siteType.setLastInput(highestIndexInputPin);
             StructList.Builder<SitePin.Builder> pins = siteType.initPins(pinNames.size());
             for(int j=0; j < pinNames.size(); j++) {
+                com.xilinx.rapidwright.device.SitePin site_pin = new com.xilinx.rapidwright.device.SitePin(site, pinNames.get(j));
```

Comment:
This could be simplified with:
```
                BELPin belPin = site.getBELPin(pinNames.get(j));
                pin.setBelpin(allBELPins.getIndex(belPin));
```

Reply:
Thanks, done.